mongo/session.go

```diff
@@ -193,9 +194,12 @@ func (s *sessionImpl) WithTransaction(ctx context.Context, fn func(sessCtx Sessi
 			default:
 			}
 
-			if cerr, ok := err.(CommandError); ok {
-				if cerr.HasErrorLabel(driver.TransientTransactionError) {
-					continue
+			currErr := err
+			for ; currErr != nil; currErr = unwrap(currErr) {
```

Comment:
Can we move this into a separate helper function? That way, we wouldn't be adding an additional loop into this function and would not need the new `RetryLoop` label, which is making this code harder to read IMO.

Reply:
done